sdk/helper/ldaputil/config.go

```diff
@@ -242,6 +248,10 @@ func NewConfigEntry(existing *ConfigEntry, d *framework.FieldData) (*ConfigEntry
 		cfg.AnonymousGroupSearch = d.Get("anonymous_group_search").(bool)
 	}
 
+	if _, ok := d.Raw["username_as_alias"]; ok || !hadExisting {
```

Comment:
Would it be better to check and error on the mutual exclusivity with `userfilter` somewhere in here? It would bubble up the error sooner on config rather than expect this behavior on login.

Reply:
This is kind of annoying for the user because `userfilter` has a default (`({{.UserAttr}}={{.Username}})`), so you have to explicitly clear it out in the config.

FWIW, I did a lot of testing and could not think of a way to get an identity collision, unless you ignore the warning about `userattr` not being set in `userfilter`. Thoughts on not having this mutual exclusivity?